static/js/draw.js

```diff
@@ -0,0 +1,446 @@
+	}
+}
+	 
+
+function Polygon(listOfDots,map,pen,color){
+	this.listOfDots = listOfDots;
+	this.map = map;
+	this.coords = new Array();
+	this.parent = pen;
+	this.des = 'Hello';
+		
+		
+	var thisObj=this;
+	$.each(this.listOfDots,function(index,value){
+		thisObj.coords.push(value.getLatLng());
+	});
+		
+	this.polygonObj= new google.maps.Polygon({
+		paths: this.coords,
+		strokeColor: "#FF0000",
+		strokeOpacity: 0.8,
+		strokeWeight: 2,
+		fillColor: "#FF0000",
+		fillOpacity: 0.35,
+		map:this.map
+	});
+		
+	this.remove = function(){
+		this.info.remove();
+		this.polygonObj.setMap(null);
+	}
+		
+	this.getContent = function(){
+		return this.des;
+	}
+		
+		
+	this.getPolygonObj= function(){
+		return this.polygonObj;
+	}
+		
+	this.getListOfDots = function (){
+		return this.listOfDots;
+	}
+		
+	this.getPlots = function(){
+		return this.polygonObj.getPaths();
+	}
+		
+	this.getColor=function(){
+		return 	this.getPolygonObj().fillColor;
+	}
+		
+	this.setColor = function(color){
+		return 	this.getPolygonObj().setOptions(
+			{fillColor:color,
+			strokeColor:color,
+			strokeWeight: 2
+			}
+		);
+	}
+		
+		
+	this.info = new Info(this,this.map);
+		
+	this.addListener = function(){
+			var info=this.info;
+			var thisPolygon=this.polygonObj;
+			google.maps.event.addListener(thisPolygon, 'rightclick', function(event) { 					
+			info.show(event.latLng);
+		});				
+	}	
+	this.addListener();
+							  
+	}
+	 	 
+function Info(polygon,map){
+	this.parent = polygon;
+	this.map = map;
+		
+	this.color =  document.createElement('input');
+		
+	this.button = document.createElement('input');
+	$(this.button).attr('type','button');
+	$(this.button).val("Change Color");
+	var thisOjb=this;
+		
+	this.changeColor= function(){
+		thisOjb.parent.setColor($(thisOjb.color).val());
+	}
+		
+	this.getContent = function(){
+		var content = document.createElement('div');
+			
+		$(this.color).val(this.parent.getColor());	
+		$(this.button).click(function(){
+			thisObj.changeColor();
+		});
+			
+		$(content).append(this.color);		
+		$(content).append(this.button);
+		return content;
+	}
+		
+	thisObj=this;
+	this.infoWidObj = new google.maps.InfoWindow({
+	 content:thisObj.getContent()
+	});
+		
+	this.show = function(latLng){
+		this.infoWidObj.setPosition(latLng);
+		this.infoWidObj.open(this.map);
+	}
+		
+	this.remove = function(){
+	 	this.infoWidObj.close();
+	}
+}
+
+function Point_string(){	
+		
+	var Point = PointArray[0];
+    	
+    PointArray.push(Point);
+    	    
+    var c_str="POLYGON((";
+    var string_append="";
+    var lng ="";
+    var lat = "";
+    for(var i=0; i<PointArray.length;i++){
+       	lng = PointArray[i].lng();
+       	lat = PointArray[i].lat();
+       	string_append+=lng+" "+lat +",";
+    };
+    c_str+=string_append.substring(string_append,string_append.length-1)+"))";
+ 	return c_str;
+}
+
+
+function Point_string_update(){    
+    var Point = curLatLng[0];
+	curLatLng.push(Point);
```

Comment:
instead of two function Point_string_update() and Point_string() use one function.
use "if condition"  of "Flag" in one of the above  function.